### PR TITLE
Bugfix/lens check if supported

### PR DESCRIPTION
**Pull request by nbfalcon**
- In `lsp--auto-configure`: add `lsp--lens-enable` instead of `lsp-lens-mode`, so that the latter won't be enabled for servers that don't support lenses. Unlike stated in the commit messages, this doesn't error. The errors were either fixed for some other reason or caused by a broken environment of mine.
- `lsp-lens-mode` can now be disabled, without the lenses reappearing shortly thereafter.

### CODE REVIEW

**Comment by nbfalcon** [comment]
This is the last PR that I'm making from upstream.

**Comment by nbfalcon** [comment]
Note that disabling `lsp-lens-mode` and then reverting the buffer will turn it back on again. This is because doing so causes `lsp-configure-buffer` to be re-run, and is not an issue that needs to be addressed (at least here, probably at all).

File: lsp-lens.el
Comment (yyoncho) [inline lsp-lens.el:367]: dont we need a check if this is nil?

Comment (nbfalcon) [reply lsp-lens.el]: It indeed can be nil. Fixed.

**Review by yyoncho — APPROVED** [review]

**Comment by nbfalcon** [comment]
Can I just merge this (given that you approved), or do you want to merge PRs yourself?

**Comment by yyoncho** [comment]
After approval you may merge it as it is or keep it open for some time if you think that more reviewers are needed for the pr.

**Comment by nbfalcon** [comment]
Rerun CI.

**Comment by nbfalcon** [comment]
@yyoncho I cannot merge this because the windows-test CI is broken (`::set_env` disabled).

**Comment by jcs090218** [comment]
The CI failed because this PR uses `v3` instead of `master`.

**Comment by nbfalcon** [comment]
@jcs090218 `v3`? How would I make it not fail?

**Comment by jcs090218** [comment]
I think this PR opened while CI is still in `v3`. Currently the master branch is using `master`. I don't knw how to get around this; yet I think there should be a way to update PR's CI environment. Or else, I don't see a solution here. 😕 If there isn't a way to do this, then I assumed this should be an enhancement for GitHub. Like updating the CI testing environment respect to the master branch, etc.

**Pull request merged**
